src/main/java/fi/aalto/cs/apluscourses/dal/APlusExerciseDataSource.java

```diff
@@ -275,5 +284,11 @@ public SubmissionResult parseSubmissionResult(@NotNull JSONObject object,
       return SubmissionResult.fromJsonObject(object, exercise);
     }
 
+    @Override
+    public String parseUserName(@NotNull JSONObject object) {
+      var fullName = object.optString("full_name");
+      var username = object.optString("username");
+      return fullName.equals("") ? username : fullName;
+    }
```

Comment:
Remark: would it be more in line with the rest of the class if this returned `User` object?